psclient/src/main/java/com/majeur/psclient/service/ShowdownService.kt

```diff
@@ -39,6 +39,21 @@ class ShowdownService : Service() {
     private var previousChatRoomId: String? = null
     private var previousBattleRoomId: String? = null
 
+    val replayManager by lazy {
+        ReplayManager(okHttpClient, object : ReplayManager.ReplayCallback {
```

Comment:
Instead of the ReplayCallback interface, you can pass directly ShodownService.this and retrieve everything you need inside ReplayManager directly from service reference (such as okHttpClient, battleRoomMessageObserver and processServerData method)